CHANGELOG.md

```diff
@@ -1,10 +1,10 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
-## \[Unreleased\]
+## \[Unreleased\] - 2023-05-16
```

Comment:
The date will be added when we know when the version is released. Otherwise you will need to change this every day your PR has not been merged ^^

Reply:
so for writing the changes, I should update the date every day otherwise not writing the date means not writing the changes and its a bad idea right ?

Reply:
A changelog is a small summary of the changes made in a project. It is designed for users to know what changed between versions so they can see if it is likely to have broken anything for them or fixed anything. The date is then useful if something stopped working at a given point in time as you can trace it back to the relevant version.

The "Unreleased" section is about things which will be in the next version. This development version is only used by people who cloned from github where we can check individual commits anyway, so the date is not necessarily useful here. Plus if every committer had to update it every time they commit, it would be really annoying and would permanently create merge conflicts.

Therefore the format for the change log is:

```
# Change Log
All notable changes to this project will be documented in this file.

## \[Unreleased\]

### Added

- #NUMBER - Completely new thing that was added in PR NUMBER

### Fixed

- #NUMBER - Thing that was broken now works!

### Changed

- I changed something (PR number may or may not be useful)

## \[VERSION NUMBER\] - VERSION RELEASE DATE

### Added

-   #1352 : Added a change log.

### Fixed

-   #1367 : Use an absolute path to link to Python library.

### Changed

-   \[INTERNALS\] I changed something only devs will notice

```

Reply:
understood, I'm confused about one thing, maybe I should write the changes in the current version part, and leave the unreleased part empty.

Reply:
What "current version part"? The current version on GitHub is the version which has not yet been versioned and released to PyPi. The current version *is* the unreleased version

Reply:
ah, true, i see, i see.